crates/examples/libp2p/all.rs

```diff
@@ -46,12 +47,18 @@ async fn main() {
             IpAddr::V4(Ipv4Addr::LOCALHOST),
             8000 + (u16::try_from(i).expect("failed to create advertise address")),
         );
+        let builder_address = Url::parse(&format!(
+            "http://localhost:{}",
+            9000 + (u16::try_from(i).expect("failed to create builder address"))
```

Comment:
Looks like we hard code this a few times. Could all of the address logic here be moved somewhere central? A single change to any of these begets a change to all of them.

Reply:
Same here as for builder_address @QuentinI , this could be updated after multi-builder thing.